### ISSUE TITLE
Planned in Brussels - filter

### ISSUE THREAD

**Issue by Sjan1** [Sjan1/r-for-proteomics-tsl#2]
The filter within `apply(etab,...)` should be ideally populated from `fcol` defined above. As it stands, we input the information on unique combinations of experimental factors twice.

```r
fcols <- c("phenotype", "treatment", "biorep")
etab <- experimentHierarchy(exp, fcols)

msnl <- apply(etab, 1, function(.etab) {
  filenames <- exp %>%
    filter(biorep == .etab[["biorep"]], 
           phenotype == .etab[["phenotype"]],
           treatment == .etab[["treatment"]]) %>%
    select(name)
```

**Comment by lgatto** [comment]
Not really. The variables are defined in the `exp` dataframe, but before continuing, you need to define which ones are relevant to define the files to be analysed together. So the `fcols` variable (the name is actually not ideal, `pvars` would be better) above is a user input, that depends on the question of interest, and that input must match some of the variables in `exp`. One could of course have a selection menu (command line or GUI) for the user to choose.

**Comment by Sjan1** [comment]
Yes, we need the user input here, which goes into fcols.

 What I meant was was, if we can generate the filter:
```r
filter(biorep == .etab[["biorep"]], 
           phenotype == .etab[["phenotype"]],
           treatment == .etab[["treatment"]]) %>%
    select(name)
```

from `fcols`?

Presuming e.g. `biorep` can be repaced with `fcols[3]`, but elegant would be to generate the whole filter itself using `length(fcols)`, without the need to alter the code, except `fcol`. (It is just an idea.)

**Comment by lgatto** [comment]
Yes, indeed, something along these lines. I will look into this when I have some time.
